dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -51,6 +53,63 @@ public static void prepare() throws Exception {
         rulesAPI   = APILocator.getRulesAPI();
     }
 
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User with permission try to delete a rule
+     * Should: Delete it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldDeleteRules() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        this.addPermission(role, host, true);
+
+        assertTrue(existRule(user, rule));
+
+        rulesAPI.deleteRule(rule, user, false);
+
+        assertFalse(existRule(user, rule));
+
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#deleteRule(Rule, User, boolean)}
+     * When: User without permission try to delete a rule
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldNotDeleteRules() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+        final Rule rule = new RuleDataGen().host(host).nextPersisted();
+
+        try {
+            rulesAPI.deleteRule(rule, user, false);
+            throw new AssertionError("DotSecurityException Expected");
+        } catch (DotSecurityException e) {
+            assertFalse(existRule(user, rule));
+        }
+    }
+
+    private boolean existRule(User user, Rule rule) throws DotDataException, DotSecurityException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5156545)